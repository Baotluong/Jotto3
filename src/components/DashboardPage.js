import React from 'react';
import { Link } from 'react-router-dom';
import { connect } from 'react-redux';
import { startSinglePlayerGame } from '../actions/game';

export class DashboardPage extends React.Component {
    constructor(props) {
        super(props);
    }
    onClick = () => {
        this.props.startSinglePlayerGame();
        this.props.history.push(`game/${this.props.userID}`);
    }
    render() {
        return (
            <div>
                Dashboard page content
                <button onClick={this.onClick}>startSinglePlayerGame</button>
                <Link to="/vssetup">
                    <h3>Versus Real Player</h3>
                </Link>
            </div>
        )
    }
}

const mapStateToProps = (state, props) => ({
    userID: state.auth.uid
});

const mapDispatchToProps = (dispatch) => ({
    startSinglePlayerGame: () => dispatch(startSinglePlayerGame()),
});

export default connect(mapStateToProps, mapDispatchToProps)(DashboardPage);