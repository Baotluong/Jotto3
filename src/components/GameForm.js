import React from 'react';

export default class GameForm extends React.Component {
    constructor(props) {
        super(props);
        this.state = {
            guess: ''
        };
    }
    onGuessChange = (e) => {
        const guess = e.target.value;
        this.setState(() => ({
            guess
        }));
    }
    onSubmit = (e) => {
        e.preventDefault();
        this.setState(() => ({
            guess: ''
        }));
        this.props.onSubmit(this.state.guess);
    }
    render() {
        return (
            <form
                onSubmit={this.onSubmit}
            >
                <input
                    type="text"
                    placeholder="Make a Guess!"
                    value={this.state.guess}
                    onChange={this.onGuessChange}
                />
                <button>Submit</button>
            </form>
        );
    }
}   