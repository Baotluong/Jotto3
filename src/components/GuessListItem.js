import React from 'react';

const GuessListItem = ({ guess }) => (
    <div>
        <h3>{guess}</h3>
    </div>
);

export default GuessListItem;