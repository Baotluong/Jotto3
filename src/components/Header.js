import React from 'react';
import { Link } from 'react-router-dom';

export const Header = ({ startLogout }) => (
  <header className="header">
    <div className="content-container">
      <div className="header__content">
        <Link className="header__title" to="/">
          <h1><span className="header__question-mark">?</span>JOTTO</h1>
        </Link>
      </div>
    </div>
  </header>
);

export default Header;