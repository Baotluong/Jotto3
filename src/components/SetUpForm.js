import React from 'react';

export const SetUpForm = () => (
    <form>
        <h3>SetUpForm</h3>
        <input type="text" />
    </form>
);

export default SetUpForm;