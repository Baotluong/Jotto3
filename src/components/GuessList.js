import React from 'react';
import { connect } from 'react-redux';
import GuessListItem from './GuessListItem';

export const GuessList = ({ guesses }) => (
    <div>
        {
            guesses.length === 0 ? (
                <div>
                    <span>No Guesses</span>
                </div>
            ) : (
                guesses.map((guess) => {
                    return <GuessListItem key={guess} guess={guess}/>;
                })
            )
        }
    </div>
);

const mapStateToProps = (state, props) => ({
    guesses: state.game[props.gameID].guesses
});

export default connect(mapStateToProps)(GuessList);