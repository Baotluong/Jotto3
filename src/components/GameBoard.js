import React from 'react';
import { connect } from 'react-redux';
import GameForm from './GameForm';
import GuessList from './GuessList';
import { addSecret, addGuess } from '../actions/game';
import { encryptor, npcSelectSecret } from '../utility/secret';
import { checkForValidGuess } from '../utility/guess';
import encryptorCreator from 'simple-encryptor';

export class GameBoard extends React.Component {
    constructor(props){
        super(props);
        this.state = {
            gameID: this.props.gameID,
            secret: '',
            guesses: ''
        }
    }
    componentDidMount() {
        this.loadGame();
    }
    loadGame = () => {
        if (!this.props.secret) {
            this.startNewGame();
        }
    }
    startNewGame = () => {
        //TODO: Make these options later.
        const difficulty = 0;
        const secret = encryptor.encrypt(npcSelectSecret(difficulty));
        this.setState(() => ({
            secret
        }));
        this.props.addSecret(this.state.gameID, secret);
    }
    onSubmit = (guess) => {
        const error = checkForValidGuess(guess, this.state.guesses);
        if (!error) {
            this.props.addGuess(this.state.gameID, guess);
            this.setState(() => ({
                guesses: [...this.state.guesses, guess]
            }));
        }
        return error;
    } 
    render () {
        return (
            <div>
                <h3>Game Board {this.props.gameID}</h3>
                <GuessList gameID={this.props.gameID} />
                <GameForm
                    onSubmit={this.onSubmit}
                />
            </div>
        );

    }
};

const mapStateToProps = (state, props) => ({
    secret: state.game[props.gameID].secret,
    gameID: props.gameID,
    guesses: state.game[props.gameID].guesses
});

const mapDispatchToProps = (dispatch, props) => ({
    addGuess: (gameID, guess) => dispatch(addGuess(gameID, guess)),
    addSecret: (gameID, secret) => dispatch(addSecret(gameID, secret))
});

export default connect(mapStateToProps, mapDispatchToProps)(GameBoard);