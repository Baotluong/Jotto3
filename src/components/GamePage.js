import React from 'react';
import GameBoard from './GameBoard';

export const GamePage = () => (
    <div>
        <h3>Game Page</h3>
        <GameBoard />
    </div>
);

export default GamePage;