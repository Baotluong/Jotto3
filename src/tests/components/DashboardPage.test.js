import React from 'react';
import { shallow } from 'enzyme';
import { DashboardPage } from '../../components/DashboardPage';

let wrapper, instance;

beforeEach(() => {
  wrapper = shallow(<DashboardPage />);
  instance = wrapper.instance();
})

class TestDashboardPage extends DashboardPage {
  constructor(props) {
    super(props);
    this.startTwoPlayerGame = this.props.startTwoPlayerGame;
  }
}

describe('DashboardPage', () => {
  test('should render DashboardPage correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });
  
  test('should call startGame(true) when startSinglePlayerGame is called', () => {
    const startGameMock = jest.fn();
    instance.startGame = startGameMock;
    instance.startSinglePlayerGame();
    expect(startGameMock).toHaveBeenLastCalledWith(true);
  });

  test('should call startGame(false) when startSinglePlayerGame is called', () => {
    const startGameMock = jest.fn();
    instance.startGame = startGameMock;
    instance.startTwoPlayerGame();
    expect(startGameMock).toHaveBeenLastCalledWith(false);
  });

  test('should call props.startGame', () => {
    const gameID = '123';
    const startGameMock = jest.fn().mockResolvedValue(gameID);
    const wrapper = shallow(<DashboardPage startGame={startGameMock} />);
    const instance = wrapper.instance();
    instance.startGame(true);
    expect(startGameMock).toHaveBeenLastCalledWith(true);
  })

  test('should call startTwoPlayerGame on button click', () => {
    const startTwoPlayerGameMock = jest.fn();
    const wrapper = shallow(<TestDashboardPage startTwoPlayerGame={startTwoPlayerGameMock} />);
    wrapper.find('.dashboard__button__versus').simulate('click');
    expect(startTwoPlayerGameMock).toHaveBeenCalled();
  })
});