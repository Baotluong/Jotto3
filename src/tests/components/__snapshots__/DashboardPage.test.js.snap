// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DashboardPage should render DashboardPage correctly 1`] = `
<div
  className="content-container"
>
  <div
    className="dashboard__buttons"
  >
    <button
      className="button dashboard__button dashboard__button__versus"
      onClick={[Function]}
    >
      (2P) Play with a Friend!
    </button>
    <button
      className="button dashboard__button"
      onClick={[Function]}
    >
      (1P) Start New Game
    </button>
    <Link
      to="/game/undefined"
    >
      <button
        className="button dashboard__button"
      >
        (1P) Continue Game
      </button>
    </Link>
  </div>
</div>
`;
